### ISSUE TITLE
Crash: `BaseTransientBottomBar.java line 146`

### ISSUE THREAD

**Issue by zeoowl-dev** [teogor/ceres#28]
Crash: `BaseTransientBottomBar.java line 146`

```txt
Caused by java.lang.reflect.InvocationTargetException
       at java.lang.reflect.Constructor.newInstance0(Constructor.java)
       at java.lang.reflect.Constructor.newInstance(Constructor.java:343)
       at android.view.LayoutInflater.createView(LayoutInflater.java:858)
       at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:1010)
       at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:965)
       at android.view.LayoutInflater.inflate(LayoutInflater.java:663)
       at android.view.LayoutInflater.inflate(LayoutInflater.java:538)
       at dev.teogor.ceres.m3.snackbar.BaseTransientBottomBar.<init>(BaseTransientBottomBar.java:36)
       at dev.teogor.ceres.m3.snackbar.Snackbar.<init>(Snackbar.java)
       at dev.teogor.ceres.m3.snackbar.Snackbar.makeInternal(Snackbar.java:88)
       at dev.teogor.ceres.m3.SnackbarBetaM3.buildSnackbar(SnackbarBetaM3.java:88)
       at dev.teogor.ceres.m3.SnackbarBetaM3.buildSnackbar$default(SnackbarBetaM3.java:88)
       at dev.teogor.ceres.m3.SnackbarBetaM3.<init>(SnackbarBetaM3.java:88)
       at dev.teogor.ceres.m3.SnackbarBetaM3.<init>(SnackbarBetaM3.java:88)
       at dev.teogor.ceres.m3.SnackbarBetaM3$Builder.prepareSnackbar(SnackbarBetaM3.java:137)
       at dev.teogor.ceres.m3.app.BaseActivityM3.processUiEvent(BaseActivityM3.java:137)
       at dev.teogor.ceres.components.app.BaseFragment.processUiEvent(BaseFragment.java:33)
       at dev.teogor.ceres.components.app.BaseFragment$$InternalSyntheticLambda$1$93e51e306c0a7d91c4b850e5dc67e12181e9dcc370125f1a29100ef6cd2df40f$2.onChanged(BaseFragment.java:33)
       at com.google.android.datatransport.runtime.scheduling.persistence.SQLiteEventStore$$ExternalSyntheticLambda5.d(R8$$SyntheticClass:30)
       at androidx.lifecycle.LiveData.considerNotify(LiveData.java:29)
       at androidx.lifecycle.LiveData.dispatchingValue(LiveData.java:56)
       at androidx.lifecycle.LiveData.setValue(LiveData.java:14)
       at androidx.lifecycle.MutableLiveData.setValue(MutableLiveData.java)
       at dev.teogor.ceres.components.events.SingleLiveEvent.setValue(SingleLiveEvent.java:6)
       at com.zeoowl.lwp.aquarium.presentation.feature.home.HomeFragment.setupObservers$lambda-0(HomeFragment.java:111)
       at com.zeoowl.lwp.aquarium.presentation.feature.home.HomeFragment$$InternalSyntheticLambda$1$d0691176a84501196567b810ae9f295ada733c275072e27364673e0325f59bb2$0.onChanged(HomeFragment.java:111)
       at androidx.lifecycle.LiveData.considerNotify(LiveData.java:29)
       at androidx.lifecycle.LiveData.dispatchingValue(LiveData.java:56)
       at androidx.lifecycle.LiveData.setValue(LiveData.java:14)
       at androidx.lifecycle.MutableLiveData.setValue(MutableLiveData.java)
       at dev.teogor.ceres.ads.view.NativeAdView.prepNativeAd$lambda-0(NativeAdView.java:71)
       at dev.teogor.ceres.ads.view.NativeAdView$$InternalSyntheticLambda$1$3a59988f7bfb7fb66125d11b40a4f22e0b10de53fdf4ff5ea0c81630a2e5acf7$0.onChanged(NativeAdView.java:71)
       at androidx.lifecycle.LiveData.considerNotify(LiveData.java:29)
       at androidx.lifecycle.LiveData.dispatchingValue(LiveData.java:56)
       at androidx.lifecycle.LiveData.setValue(LiveData.java:14)
       at androidx.lifecycle.MutableLiveData.setValue(MutableLiveData.java)
       at androidx.lifecycle.LiveData$1.run(LiveData.java:18)
       at android.os.Handler.handleCallback(Handler.java:938)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at android.os.Looper.loopOnce(Looper.java:226)
       at android.os.Looper.loop(Looper.java:313)
       at android.app.ActivityThread.main(ActivityThread.java:8751)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1135)
```

```txt
Caused by java.lang.IllegalArgumentException: dev.teogor.ceres.m3.snackbar.Snackbar.SnackbarLayout requires a value for the com.zeoowl.lwp.aquarium:attr/colorSurface attribute to be set in your app theme. You can either set the attribute in your theme or update your theme to inherit from Theme.MaterialComponents (or a descendant).
       at com.google.android.material.resources.MaterialAttributes.resolveTypedValueOrThrow(MaterialAttributes.java:32)
       at com.google.android.material.resources.MaterialAttributes.resolveTypedValueOrThrow(MaterialAttributes.java:17)
       at com.google.android.material.color.MaterialColors.getColor(MaterialColors.java:17)
       at com.google.android.material.color.MaterialColors.layer(MaterialColors.java:146)
       at dev.teogor.ceres.m3.snackbar.BaseTransientBottomBar$SnackbarBaseLayout.createThemedBackground(BaseTransientBottomBar.java:146)
       at dev.teogor.ceres.m3.snackbar.BaseTransientBottomBar$SnackbarBaseLayout.<init>(BaseTransientBottomBar.java:146)
       at dev.teogor.ceres.m3.snackbar.Snackbar$SnackbarLayout.<init>(Snackbar.java)
       at java.lang.reflect.Constructor.newInstance0(Constructor.java)
       at java.lang.reflect.Constructor.newInstance(Constructor.java:343)
       at android.view.LayoutInflater.createView(LayoutInflater.java:858)
       at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:1010)
       at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:965)
       at android.view.LayoutInflater.inflate(LayoutInflater.java:663)
       at android.view.LayoutInflater.inflate(LayoutInflater.java:538)
       at dev.teogor.ceres.m3.snackbar.BaseTransientBottomBar.<init>(BaseTransientBottomBar.java:36)
       at dev.teogor.ceres.m3.snackbar.Snackbar.<init>(Snackbar.java)
       at dev.teogor.ceres.m3.snackbar.Snackbar.makeInternal(Snackbar.java:88)
       at dev.teogor.ceres.m3.SnackbarBetaM3.buildSnackbar(SnackbarBetaM3.java:88)
       at dev.teogor.ceres.m3.SnackbarBetaM3.buildSnackbar$default(SnackbarBetaM3.java:88)
       at dev.teogor.ceres.m3.SnackbarBetaM3.<init>(SnackbarBetaM3.java:88)
       at dev.teogor.ceres.m3.SnackbarBetaM3.<init>(SnackbarBetaM3.java:88)
       at dev.teogor.ceres.m3.SnackbarBetaM3$Builder.prepareSnackbar(SnackbarBetaM3.java:137)
       at dev.teogor.ceres.m3.app.BaseActivityM3.processUiEvent(BaseActivityM3.java:137)
       at dev.teogor.ceres.components.app.BaseFragment.processUiEvent(BaseFragment.java:33)
       at dev.teogor.ceres.components.app.BaseFragment$$InternalSyntheticLambda$1$93e51e306c0a7d91c4b850e5dc67e12181e9dcc370125f1a29100ef6cd2df40f$2.onChanged(BaseFragment.java:33)
       at com.google.android.datatransport.runtime.scheduling.persistence.SQLiteEventStore$$ExternalSyntheticLambda5.d(R8$$SyntheticClass:30)
       at androidx.lifecycle.LiveData.considerNotify(LiveData.java:29)
       at androidx.lifecycle.LiveData.dispatchingValue(LiveData.java:56)
       at androidx.lifecycle.LiveData.setValue(LiveData.java:14)
       at androidx.lifecycle.MutableLiveData.setValue(MutableLiveData.java)
       at dev.teogor.ceres.components.events.SingleLiveEvent.setValue(SingleLiveEvent.java:6)
       at com.zeoowl.lwp.aquarium.presentation.feature.home.HomeFragment.setupObservers$lambda-0(HomeFragment.java:111)
       at com.zeoowl.lwp.aquarium.presentation.feature.home.HomeFragment$$InternalSyntheticLambda$1$d0691176a84501196567b810ae9f295ada733c275072e27364673e0325f59bb2$0.onChanged(HomeFragment.java:111)
       at androidx.lifecycle.LiveData.considerNotify(LiveData.java:29)
       at androidx.lifecycle.LiveData.dispatchingValue(LiveData.java:56)
       at androidx.lifecycle.LiveData.setValue(LiveData.java:14)
       at androidx.lifecycle.MutableLiveData.setValue(MutableLiveData.java)
       at dev.teogor.ceres.ads.view.NativeAdView.prepNativeAd$lambda-0(NativeAdView.java:71)
       at dev.teogor.ceres.ads.view.NativeAdView$$InternalSyntheticLambda$1$3a59988f7bfb7fb66125d11b40a4f22e0b10de53fdf4ff5ea0c81630a2e5acf7$0.onChanged(NativeAdView.java:71)
       at androidx.lifecycle.LiveData.considerNotify(LiveData.java:29)
       at androidx.lifecycle.LiveData.dispatchingValue(LiveData.java:56)
       at androidx.lifecycle.LiveData.setValue(LiveData.java:14)
       at androidx.lifecycle.MutableLiveData.setValue(MutableLiveData.java)
       at androidx.lifecycle.LiveData$1.run(LiveData.java:18)
       at android.os.Handler.handleCallback(Handler.java:938)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at android.os.Looper.loopOnce(Looper.java:226)
       at android.os.Looper.loop(Looper.java:313)
       at android.app.ActivityThread.main(ActivityThread.java:8751)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1135)
```

**Comment by teogor** [comment]
Thank you for your feedback, @zeoowl-dev. Currently this library is in ALPHA therefore we are constantly trying to improve the performance.

**Comment by teogor** [comment]
closed #38
